extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java

```diff
@@ -1058,6 +721,366 @@ A more full example of generated client (with sub-resource) can is at the bottom
+                        handleHeaderMethod.returnValue(invocationBuilderRef);
+                        invocationBuilderEnrichers.put(handleHeaderDescriptor,
+                                subMethodCreator.getMethodParam(paramIdx));
+                    } else if (param.parameterType == ParameterType.COOKIE) {
+                        // cookies are added at the invocation builder level
+                        MethodDescriptor handleCookieDescriptor = MethodDescriptor.ofMethod(subName,
+                                subMethod.getName() + "$$" + subMethodIndex + "$$handleCookie$$" + paramIdx,
+                                Invocation.Builder.class,
+                                Invocation.Builder.class, param.type);
+                        MethodCreator handleCookieMethod = sub.getMethodCreator(handleCookieDescriptor);
+
+                        AssignableResultHandle invocationBuilderRef = handleCookieMethod
+                                .createVariable(Invocation.Builder.class);
+                        handleCookieMethod.assign(invocationBuilderRef, handleCookieMethod.getMethodParam(0));
+                        addCookieParam(handleCookieMethod, invocationBuilderRef, param.name,
+                                handleCookieMethod.getMethodParam(1),
+                                param.type,
+                                handleCookieMethod.readInstanceField(clientField, handleCookieMethod.getThis()));
+                        handleCookieMethod.returnValue(invocationBuilderRef);
+                        invocationBuilderEnrichers.put(handleCookieDescriptor,
+                                subMethodCreator.getMethodParam(paramIdx));
+                    } else if (param.parameterType == ParameterType.FORM) {
+                        formParams = createIfAbsent(subMethodCreator, formParams);
+                        subMethodCreator.invokeInterfaceMethod(MULTIVALUED_MAP_ADD, formParams,
+                                subMethodCreator.load(param.name),
+                                subMethodCreator.getMethodParam(paramIdx));
+                    } else if (param.parameterType == ParameterType.MULTI_PART_FORM) {
+                        if (multipartForm != null) {
+                            throw new IllegalArgumentException("MultipartForm data set twice for method "
+                                    + jandexSubMethod.declaringClass().name() + "#" + jandexSubMethod.name());
+                        }
+                        multipartForm = createMultipartForm(subMethodCreator,
+                                subMethodCreator.getMethodParam(paramIdx),
+                                jandexSubMethod.parameters().get(paramIdx), index);
+                    }
+
+                }
+
+                if (subMethod.getHttpMethod() == null) {
+                    // finding corresponding jandex method, used by enricher (MicroProfile enricher stores it in a field
```

Comment:
This is the only change that was entered in this PR. The rest is a copy of what we had but in the new method `handleSubResourceMethod`.